libdino/src/entity/account.vala

```diff
@@ -73,6 +69,38 @@ public class Account : Object {
         notify.connect(on_update);
     }
 
+    public async string? get_password() {
+#if WITH_SECRET
+        if (password != "") {
```

Comment:
Please invert this check, it will hopefully be more likely for the password to come from the secret service than from the db.  You can then drop the else clause and have this part one indent level lower.